matchms/similarity/spectrum_similarity_functions.py

```diff
@@ -5,69 +5,111 @@
 
 
 @numba.njit
-def collect_peak_pairs(spec1, spec2, tolerance, shift=0,
-                       mz_power=0.0, intensity_power=1.0):
+def collect_peak_pairs(spec1: numpy.ndarray, spec2: numpy.ndarray,
+                       tolerance: float, shift: float = 0, mz_power: float = 0.0,
+                       intensity_power: float = 1.0):
     # pylint: disable=too-many-arguments
     """Find matching pairs between two spectra.
 
     Args
     ----
-    spec1: numpy array
+    spec1:
         Spectrum peaks and intensities as numpy array.
-    spec2: numpy array
+    spec2:
         Spectrum peaks and intensities as numpy array.
-    tolerance : float
+    tolerance
         Peaks will be considered a match when <= tolerance appart.
-    shift : float, optional
+    shift
         Shift spectra peaks by shift. The default is 0.
-    mz_power: float, optional
+    mz_power:
         The power to raise mz to in the cosine function. The default is 0, in which
         case the peak intensity products will not depend on the m/z ratios.
-    intensity_power: float, optional
+    intensity_power:
         The power to raise intensity to in the cosine function. The default is 1.
 
     Returns
     -------
-    matching_pairs : list
-        List of found matching peaks.
+    matching_pairs : numpy array
```

Comment:
A normal array is returned. 
```suggestion
    matching_pairs : array
```

Reply:
Or let it return a numpy array. Then the `matching_pairs = numpy.array(matching_pairs)` elsewhere is no longer needed

Reply:
Unfortunately Numba complained when I did ``matching_pairs = numpy.array(matching_pairs)`` but I will check again.

Reply:
Ah, numba probably does not like a var changing types, using different var name might help

Reply:
I think it is rather something with the different types what are within the list... but not sure.
The error is ``TypingError: Invalid use of Function(<intrinsic range_iter_len>) with argument(s) of type(s): (enumerate(iter(list(int64))))``

Reply:
OK, seems to work if I do ``return numpy.array(matching_pairs.copy())``.

Reply:
I changed this and removed the now redundant ``matching_pairs = numpy.array(matching_pairs)`` lines.